### ISSUE TITLE
Two-way Data-binding with HTTP Get CORS / ASYNC

### ISSUE THREAD

**Issue by benoror** [networknt/angular-schema-form-ui-select#10]
Setting a value in $scope.model should be reflected after fetching data via `http_get` or `async` Example:

	  $scope.form = [
		     ///...
		     {
		       key: 'asyncselect',
		       options: {
		           async: {
		               call: $scope.refreshSelectAsync,
		               url : "/test/testdata.json"
		           }
		       }
		     },
		     ///...
		  ];

        //..

	  $scope.model = {
	    async: 'json-value1'
	  };

**Comment by stevehu** [comment]
Hi Ben,

Anything in the app scope should be removed and I am working on it. Some
features merged from another developer are not working and I am in a
process to remove it. Please give me some time to fix it and then you can
fork it and submit pull requests. I don't want you to waste time working on
a copy which is work in process.

Thanks,

Steve

**Comment by benoror** [comment]
Sure,

Let me know if you need help!

P.S.: I'll let you know of our own project, you guys might be interested (:

Regards!

**Comment by benoror** [comment]
Any updates regarding this? Maybe related to https://github.com/networknt/angular-schema-form-ui-select/issues/15 ?

**Comment by benoror** [comment]
@stevehu , I recently found you created another repository ([angular-schema-form-uiselect](https://github.com/networknt/angular-schema-form-uiselect)), does that mean this repo will remain unmaintained?

Also you've been discusing about using strap-select ([angular-schema-form-strapselect](https://github.com/networknt/angular-schema-form-strapselect)) instead of ui-select for Ajax calls. I would like to avoid mixing AngularUI & AngularStrap in the same app. What do you think?

Cheers!

**Comment by stevehu** [comment]
@benoror, Yes. I am planning to move to uiselect but it has an issue with tags at the moment. If you could, please fix the issue in that repo. In the new repo, each uiselect field will have its own controller and it should be easy to add logic for cascade select but it might be hard to make it flexible and configurable. I am so sorry that I couldn't allocate any time to work on it as my focus is on the API server these days. Thanks.

**Comment by benoror** [comment]
Great steve!

Unfortunately I just tried [uiselect](https://github.com/networknt/angular-schema-form-uiselect), but it only renders an input field, not a select.

**Comment by benoror** [comment]
I will fork your original ui-select and try to correct the issues myself, maybe we can merge all of them later with the enhancements. 

I'll keep you posted.

Cheers!

**Comment by stevehu** [comment]
Thanks a lot.

**Comment by benoror** [comment]
It looks that data binding is wrong. According to the [extending guide](https://github.com/Textalk/angular-schema-form/blob/master/docs/extending.md), **$$value$$** should be used/

I made a simple comparison in index.html [you can check it out here](https://github.com/benoror/angular-schema-form-ui-select/commit/257c466480c563e1fc78f5b568346f1d9e6423be)

Does it makes sense to you?

**Comment by benoror** [comment]
Update: I fixed an issue with the HTTP promise, but you can still gent the idea:

https://github.com/benoror/angular-schema-form-ui-select/commit/ae81f051094837e4f8ff20e35691551a9c801bdf

**Comment by benoror** [comment]
**Update**: I fixed it for **single-select** using $$value$$ string replacement.

Commit: https://github.com/benoror/angular-schema-form-ui-select/commit/f14b130d260d9e851b314e0cb4ed6872033ca689

Code is still dirty and unorganized. I am thinking about starting a new project from scratch:

* using unit-tests
* supporting single-select at first
* then progressively add functionality like tagging, multiselect, etc.

Would you be interested in combining efforts?
